src/lib/select/select.ts

```diff
@@ -413,9 +413,17 @@ export class MdSelect implements AfterContentInit, ControlValueAccessor, OnDestr
 
   /** The value displayed in the trigger. */
   get triggerValue(): string {
-    return this.multiple ?
-      this._selectionModel.selected.map(option => option.viewValue).join(', ') :
-      this._selectionModel.selected[0].viewValue;
+    if (this._multiple) {
+      let array = this._selectionModel.selected.map(option => option.viewValue);
+
+      if (this._isRtl()) {
+        array.reverse();
+      }
+
+      return array.join(', ');
```

Comment:
Just noticed: I think this delimiter should be configurable, as a comma may not be the right character for all locales. Add a TODO?